pyxem/signals/diffraction2d.py

```diff
@@ -499,6 +499,7 @@ def decomposition(self, *args, **kwargs):
         super(Signal2D, self).decomposition(*args, **kwargs)
         self.learning_results.loadings = np.nan_to_num(
             self.learning_results.loadings)
+        self.__class__ = Diffraction2D
```

Comment:
Could this method be changed to take the same form as all the others in pyxem? It's legacy code that may not be very good!

Reply:
Made the suggest change in latest commit.

Reply:
Thanks!